spec/knapsack_pro/config/env_spec.rb

```diff
@@ -630,26 +901,32 @@
     context 'when ENV exists' do
       context 'when KNAPSACK_PRO_FIXED_QUEUE_SPLIT=false' do
         [
-          ['AppVeyor', { 'APPVEYOR' => '123' }],
-          ['Buildkite', { 'BUILDKITE' => 'true' }],
-          ['CircleCI', { 'CIRCLECI' => 'true' }],
-          ['Cirrus CI', { 'CIRRUS_CI' => 'true' }],
-          ['Codefresh', { 'CF_BUILD_ID' => '123' }],
-          ['Codeship', { 'CI_NAME' => 'codeship' }],
-          ['GitHub Actions', { 'GITHUB_ACTIONS' => 'true' }],
-          ['GitLab CI', { 'GITLAB_CI' => 'true' }],
-          ['Heroku CI', { 'HEROKU_TEST_RUN_ID' => '123' }],
-          ['Semaphore CI 1.0', { 'SEMAPHORE_BUILD_NUMBER' => '123' }],
-          ['Semaphore CI 2.0', { 'SEMAPHORE' => 'true', 'SEMAPHORE_WORKFLOW_ID' => '123' }],
-          ['Travis CI', { 'TRAVIS' => 'true' }],
-          ['Unsupported CI', {}],
-        ].each do |ci, env|
+          ['AppVeyor', { 'APPVEYOR' => '123' }, false],
+          ['Buildkite', { 'BUILDKITE' => 'true' }, true],
+          ['CircleCI', { 'CIRCLECI' => 'true' }, false],
+          ['Cirrus CI', { 'CIRRUS_CI' => 'true' }, false],
+          ['Codefresh', { 'CF_BUILD_ID' => '123' }, false],
+          ['Codeship', { 'CI_NAME' => 'codeship' }, true],
+          ['GitHub Actions', { 'GITHUB_ACTIONS' => 'true' }, true],
+          ['GitLab CI', { 'GITLAB_CI' => 'true' }, true],
+          ['Heroku CI', { 'HEROKU_TEST_RUN_ID' => '123' }, false],
+          ['Semaphore CI 1.0', { 'SEMAPHORE_BUILD_NUMBER' => '123' }, false],
+          ['Semaphore CI 2.0', { 'SEMAPHORE' => 'true', 'SEMAPHORE_WORKFLOW_ID' => '123' }, false],
+          ['Travis CI', { 'TRAVIS' => 'true' }, true],
+          ['Unsupported CI', {}, true],
+        ].each do |ci, env, ci_env|
           it "on #{ci} it is false" do
+            stub_const("ENV", env.merge({ 'KNAPSACK_PRO_FIXED_QUEUE_SPLIT' => 'false' }))
+
             logger = instance_double(Logger)
             allow(KnapsackPro).to receive(:logger).and_return(logger)
-            expect(logger).not_to receive(:info)
-
-            stub_const("ENV", env.merge({ 'KNAPSACK_PRO_FIXED_QUEUE_SPLIT' => 'false' }))
+            if ci_env == false
```

Comment:
Have you considered going with the following and removing `ci_env`? (Same for the test below.)

```ruby
            if described_class.detected_ci.new.fixed_queue_split == false
```